std/machines/hash/poseidon_bb.asm

```diff
@@ -98,26 +99,11 @@ machine PoseidonBB(mem: Memory, split_bb: SplitBB) with
     CLK[STATE_SIZE] * (high_addr - output_addr_high) = 0;
     CLK[STATE_SIZE] * (low_addr - output_addr_low) = 0;
 
-    // How far away from overflowing the low limb is:
-    let low_diff = low_addr - (0x10000 - 4);
-    // Is one if low limb is about to overflow:
-    let low_overflow;
-    low_overflow = 1 - low_diff_inv * low_diff;
-    // Helper to allow low_overflow to be boolean
-    let low_diff_inv;
-    // Ensures that (low_diff_inv * low_diff) is boolean,
-    // and that low_diff_inv is not 0 when low_diff is not 0:
-    (low_diff_inv * low_diff - 1) * low_diff = 0;
-
-    // Increment the low limb if address is not being set:
-    (1 - is_addr_set') * (
-        // If low limb is about to overflow, next value must be 0:
-        low_overflow * low_addr' +
-        // Otherwise, next value is current plus 4:
-        (1 - low_overflow) * (low_addr + 4 - low_addr')
-    ) = 0;
-    // Set high limb, incremented if low overflowed:
-    (1 - is_addr_set') * (high_addr' - high_addr - low_overflow) = 0;
+    // Increment the address in every row but the ones it is set.
+    std::array::map(
+        word_increment_ptr(high_addr, low_addr, high_addr', low_addr'),
+        |c| std::constraints::make_conditional(c, (1 - is_addr_set'))
```

Comment:
Now I regret putting the condition as second parameter. Maybe we can change that in a future PR.